File: src/Goods.BackOffice/client-app/src/domain/vehicles/vehicle.ts
import { Page } from "../../tools/types/page";
import { VehicleCategory } from "./enums/vehicleCategory";

export class Vehicle {
  constructor(
    public readonly id: string,
    public readonly driverId: string | null,
    public readonly name: string,
    public readonly stateNumber: string,
    public readonly vehicleCategory: VehicleCategory,
    public readonly averageSpeed: number,
    public readonly fuelConsumption: number,
  ) {}
}

export function mapToVehiclesPage(data: any): Page<Vehicle> {
  return Page.convert(data, mapToVehicle);
}

export function mapToVehicles(data: any[]): Vehicle[] {
  return data.map(mapToVehicle);
}

export function mapToVehicle(data: any): Vehicle {
  return new Vehicle(
    data.id,
    data.driverId ?? null,
    data.name,
    data.stateNumber,
    data.vehicleCategory,
    data.averageSpeed,
    data.fuelConsumption,
  );
}

File: src/Goods.Domain/Vehicles/Vehicle.cs
using Goods.Domain.Vehicles.Enums;

namespace Goods.Domain.Vehicles;

public class Vehicle(
    Guid id, 
    Guid? driverId,
    String name,
    String stateNumber, 
    VehicleCategory vehicleCategory,
    Decimal averageSpeed,
    Decimal fuelConsumption,
    Boolean isRemoved)
{
    public Guid Id { get; } = id;
    public Guid? DriverId { get; } = driverId;
    public String Name { get; } = name;
    public String StateNumber { get; } = stateNumber;
    public VehicleCategory VehicleCategory { get; } = vehicleCategory;
    public Decimal AverageSpeed { get; } = averageSpeed;
    public Decimal FuelConsumption { get; } = fuelConsumption;
    public Boolean IsRemoved { get; } = isRemoved;
}
